src/input/input.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace GOL\Input;
+
+use GOL\Boards\Board;
+use Ulrichsg\Getopt;
+
+/**
+ * Interface for pluggable inputs.
+ *
+ * Use prepareBoard() to prepare a Board and register()
+ * to register optional arguments.

```

Comment:
When documenting a base-class it is usually better to target the implementors of the class, than the users of the class. So something like "Implement prepareBoard() to fill the input like your input requires it", etc, would have been better.